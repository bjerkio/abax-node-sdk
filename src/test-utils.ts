import { MockAgent, type Interceptable, setGlobalDispatcher } from 'undici';
import { AbaxClient } from './abax-client';

export function initialiseClientAndMockPool(): {
  client: AbaxClient;
  mockPool: Interceptable;
} {
  const mockAgent = new MockAgent();
  mockAgent.disableNetConnect();
  setGlobalDispatcher(mockAgent);

  const mockPool = mockAgent.get('https://api.abax.cloud');

  const client = new AbaxClient({
    baseUrl: 'production',
    apiKey: 'kanonball!',
  });

  return { client, mockPool };
}
