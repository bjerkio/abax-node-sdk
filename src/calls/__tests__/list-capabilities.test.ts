import { describe, expect, it } from 'vitest';
import { initialiseClientAndMockPool } from '../../test-utils.js';

describe('list capabilities', () => {
  it('should list capabilitites', async () => {
    const { client, mockPool } = initialiseClientAndMockPool();

    mockPool
      .intercept({
        path: '/v1/api-capabilities',
        method: 'GET',
      })
      .reply(200, {
        items: [
          {
            capability: 'query-equipment',
          },
          {
            capability: 'query-equipment-locations',
          },
          {
            capability: 'query-people',
          },
        ],
      })
      .times(1);
    const capabilites = await client.listCapabilities();

    expect(capabilites).toMatchSnapshot();
  });
});
