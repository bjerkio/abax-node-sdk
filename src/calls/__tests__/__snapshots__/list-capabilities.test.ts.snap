// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`list capabilities > should list capabilitites 1`] = `
{
  "items": [
    {
      "capability": "query-equipment",
    },
    {
      "capability": "query-equipment-locations",
    },
    {
      "capability": "query-people",
    },
  ],
}
`;
